src/test/java/com/example/demo/services/WishServiceTests.java

```diff
@@ -0,0 +1,189 @@
+package com.example.demo.services;
+
+import org.easymock.Capture;
+import org.easymock.CaptureType;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockRunner;
+import org.easymock.EasyMockSupport;
+import org.easymock.IAnswer;
+import org.easymock.Mock;
+import org.easymock.TestSubject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.example.demo.entities.Category;
+import com.example.demo.entities.Item;
+import com.example.demo.entities.Subcategory;
+import com.example.demo.entities.User;
+import com.example.demo.entities.Wish;
+import com.example.demo.exceptions.AuctionAppException;
+import com.example.demo.exceptions.InvalidDataException;
+import com.example.demo.exceptions.UnallowedOperationException;
+import com.example.demo.models.ItemModel;
+import com.example.demo.models.WishModel;
+import com.example.demo.repositories.ItemsRepository;
+import com.example.demo.repositories.UsersRepository;
+import com.example.demo.repositories.WishRepository;
+
+import static org.easymock.EasyMock.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RunWith(EasyMockRunner.class)
+public class WishServiceTests extends EasyMockSupport{
+
+	@Mock
+	WishRepository wishesRepoMock;
+	@Mock
+	ItemsRepository itemsRepoMock;
+	@Mock
+	UsersRepository usersRepoMock;
+	
+	@TestSubject
+	WishService wishService=new WishService(wishesRepoMock,itemsRepoMock,usersRepoMock);
+	
+	@Test(expected = UnallowedOperationException.class)
+	public void testCreateWishOwnItemShouldThrowException() throws AuctionAppException {
+		WishModel wish=new WishModel();
+		wish.setItem(new ItemModel());
+		User user=new User();
+		List<Item> items=new ArrayList<Item>();
+		items.add(new Item());
+		user.setItems(items);
+		expect(usersRepoMock.findById(anyInt())).andReturn(Optional.of(user));
+		replayAll();
+		
+		wishService.createWish(wish, user);
+		
+		verifyAll();		
```

Comment:
This one is testing if the method throws a proper exception

Reply:
oh, then we need to wrap this in try {} finally { verifyAll(); }

because

when exception is thrown, methods stops executing, and verify all is not called

in try call method that throws exception